spec/utils_spec.lua

```diff
@@ -92,15 +92,53 @@ describe("utils.isValidAOAddress", function()
 end)
 
 describe("utils.validateUndername", function()
-	it("should throw an error for invalid undernames", function()
-		local invalid, error = pcall(utils.validateUndername, "_invalid_undername_")
-		assert.is_false(invalid)
-		assert.is_not_nil(error)
-	end)
-
-	it("should not throw an error for a valid undername", function()
-		local valid, error = pcall(utils.validateUndername, "valid-undername-123")
-		assert.is_true(valid)
-		assert.is_nil(error)
+	it("should allow valid undernames", function()
+		local validNames = {
+			"@",
+			"a",
+			"aA",
+			"Z_",
+			"z-",
+			"1",
+			"1-",
+			string.rep("a", 61),
+			string.rep("z", 60) .. "_",
+			string.rep("0", 60) .. "-",
+		}
+
+		for _, name in ipairs(validNames) do
+			local success, error = pcall(utils.validateUndername, name)
+			assert.is_true(success, name .. " was invalid")
+			assert.is_nil(error, name .. " got an error")
```

Comment:
Might want to print the error string

Reply:
https://github.com/ar-io/ar-io-ant-process/pull/60/commits/ba64ba4467d5f0b7fbbef25a5cfebd1bd4153d7b